### PR TITLE
On macOS, add documentEdited APIs

### PR DESCRIPTION
**Pull request by amrbashir**
- [x] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

Port of https://github.com/tauri-apps/tao/commit/33fdeab6291d4aef8ea9facb58fe9583f6c1aaf3

### CODE REVIEW

**Comment by kchibisov** [comment]
How does it work? Is it basically enable/disable prompt and there's no event attached to it that says that user is trying to close the window with edit? Do you get `CloseRequested` in that case?

Is that to prevent force close by the system, since winit app can ignore `CloseRequested` and emulate the dialog thingy?

**Comment by amrbashir** [comment]
I don't use macOS but I think it is just a hint to the user that this window has unsaved changes, see this image 
image

File: src/platform/macos.rs [hunk 1]
```diff
@@ -36,6 +36,12 @@ pub trait WindowExtMacOS {
 
     /// Sets whether or not the window has shadow.
     fn set_has_shadow(&self, has_shadow: bool);
+
+    /// Get the window's edit state.
+    fn is_document_edited(&self) -> bool;
+
+    /// Put the window in a state which indicates a file save is required.
+    fn set_decoument_edited(&self, edited: bool);
```
Comment (madsmtm) [inline src/platform/macos.rs:44]: Typo
```suggestion
    fn set_document_edited(&self, edited: bool);
```

File: src/platform/macos.rs [hunk 2]
```diff
@@ -36,6 +36,12 @@ pub trait WindowExtMacOS {
 
     /// Sets whether or not the window has shadow.
     fn set_has_shadow(&self, has_shadow: bool);
+
+    /// Get the window's edit state.
```
Comment (madsmtm) [inline src/platform/macos.rs:40]: Would be nice to document how this relates to `CloseRequested`, something like:
```suggestion
    /// Get the window's edit state.
    ///
    /// # Examples
    ///
    /// ```ignore
    /// WindowEvent::CloseRequested => {
    ///     if window.is_document_edited() {
    ///         // Show the user a save pop-up or similar
    ///     } else {
    ///         // Close the window
    ///         drop(window);
    ///     }
    /// }
    /// ```
```

**Review by madsmtm — CHANGES_REQUESTED** [review]
Tested it, it has no effect on `CloseRequested`, is simply a visual indication (though that would be nice to document!). It looks as follows on macOS 10.14.6:

Normal:
unedited

Edited:
edited

**Review by madsmtm — APPROVED** [review]
Thanks, will merge in a few hours once CI catches up

**Pull request merged**